### PR TITLE
[KYUUBI #2011] Introduce to very basic hive engine

### PR DESCRIPTION
**Pull request by yaooqinn**
### _Why are the changes needed?_

Support a lightweight hive engine which has some advantages over hiveserver2
- separate engine/process to the multi-tenancy framework for better stability, without having a heavy server
- more engines introduced in kyuubi, hive may be aged but still widely used
- also good for us to do some cross verifications for spark/flink etc with a stable hive.

In this PR, a basic hive engine backend is introduced.
- only engine side
- only execute statement is supported

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @ulysses-you @pan3793 @SteNicholas @turboFei @yanghua @hddong

File: externals/kyuubi-hive-sql-engine/src/main/scala/org/apache/kyuubi/engine/hive/session/HiveSessionManager.scala [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.engine.hive.session
+
+import java.io.File
+import java.util.concurrent.Future
+
+import org.apache.hadoop.hive.conf.HiveConf
+import org.apache.hive.service.cli.{SessionHandle => SH}
```
Comment (ulysses-you) [inline externals/kyuubi-hive-sql-engine/src/main/scala/org/apache/kyuubi/engine/hive/session/HiveSessionManager.scala:24]: nit: `SH` -> `ImportedSessionHandle`

File: externals/kyuubi-hive-sql-engine/src/main/scala/org/apache/kyuubi/engine/hive/HiveBackendService.scala [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.engine.hive
+
+import org.apache.kyuubi.engine.hive.session.HiveSessionManager
+import org.apache.kyuubi.service.{AbstractBackendService, Serverable}
+import org.apache.kyuubi.session.SessionManager
+
+class HiveBackendService(serv: Serverable)
```
Comment (ulysses-you) [inline externals/kyuubi-hive-sql-engine/src/main/scala/org/apache/kyuubi/engine/hive/HiveBackendService.scala:24]: shall we pass `hiveEngine: HiveSQLEngine`  for more code readable ?

File: externals/kyuubi-hive-sql-engine/pom.xml [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  ~ limitations under the License.
+  -->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>kyuubi-parent</artifactId>
+        <groupId>org.apache.kyuubi</groupId>
+        <version>1.5.0-SNAPSHOT</version>
```
Comment (ulysses-you) [inline externals/kyuubi-hive-sql-engine/pom.xml:25]: now is 1.6.0-SNAPSHOT

File: externals/kyuubi-hive-sql-engine/src/main/scala/org/apache/kyuubi/engine/hive/HiveBackendService.scala (hunk 2 above)
Comment (yaooqinn) [reply externals/kyuubi-hive-sql-engine/src/main/scala/org/apache/kyuubi/engine/hive/HiveBackendService.scala]: addressed

File: externals/kyuubi-hive-sql-engine/src/main/scala/org/apache/kyuubi/engine/hive/session/HiveSessionManager.scala (hunk 1 above)
Comment (yaooqinn) [reply externals/kyuubi-hive-sql-engine/src/main/scala/org/apache/kyuubi/engine/hive/session/HiveSessionManager.scala]: looks good, nice catch

**Review by ulysses-you — APPROVED** [review]

**Comment by yaooqinn** [comment]
thanks merged to master

**Pull request closed without merging**